### PR TITLE
Add message with link to docs for fields without any filter widgets

### LINKED ISSUE #10334: Field filters on columns with no available widgets
When you create a new field filter in the SQL editor, you're able to pick which field to map it to from all fields in the current database.

However, if it's a type where we don't have any available widgets, that "Filter widget type" drop down is hidden. (Ratings here is a "Score" rather than "Category" in the datamodel.)

We document this behavior [here](https://metabase.com/docs/latest/users-guide/13-sql-parameters.html):
> Note: If you’re not seeing the option to display a filter widget, make sure the mapped field is set to one of the above types, and then try manually syncing your database from the Databases section of the Admin Panel to force Metabase to scan and cache the field’s values.

Even though we document it, it's confusing. I think this issue is expected behavior according to the docs: https://github.com/metabase/metabase/issues/8800

Some options:
1. Don't show every field in the "Field to map to" dropdown. Only show fields that have available widgets.
2. Let users select any field, but alert them that there are no available widgets. This could just be displaying a message instead of the dropdown.
3. Add more filter widgets. If we add a few more filter widgets (text, number, etc), it seems like every type could have a corresponding widget.

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #10334 (kinda)

Of the options outlined in that issue, this PR implements (2). It's a quick fix that will hopefully help alleviate some confusion. Thanks @flamber for weighing in on that issue!

When a field has a valid filter it looks like before:
image
Otherwise, there's a message with a link to the docs:
image

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx [hunk 1]
```diff
@@ -173,10 +176,26 @@ export default class TagEditorParam extends Component {
...
+                another field or`}{" "}
+                <Link
+                  to={MetabaseSettings.docsUrl(
+                    "users-guide/13-sql-parameters",
+                    "the-field-filter-variable-type",
+                  )}
+                  target="_blank"
+                  className="link"
+                >
+                  {t`read about this issue in the docs.`}
```
Comment (paulrosenzweig) [inline frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx]: Splitting the sentence like this and putting part of it in the link probably makes it hard to create a translation cleanly. Is that an issue? Is there some way around that or should I just change the copy?

Comment (tlrobinson) [reply frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx]: We don't have a better way to handle this currently. Probably best to do it like this though:

```
{t`Oh no! There are no filter widgets for this field. Select another field or ${
  <Link
    to={MetabaseSettings.docsUrl("users-guide/13-sql-parameters", "the-field-filter-variable-type")}
    target="_blank"
    className="link"
  >
    {t`read about this issue in the docs`}
  </Link>
}.`}
```

**Review by tlrobinson — APPROVED** [review]
@kdoh @mazameli  can you verify the copy looks good

**Review by mazameli — CHANGES_REQUESTED** [review]
This isn't quite right. Pretty sure there are valid cases where you would want to set up a field filter without a widget, right? The docs even say, "you can select `None` if you don’t want a widget at all." 

Maybe the move is, for field types that don't have any widgets, to display the widget dropdown with "none" selected, with the dropdown disabled, and with text below it, similar to what you have but smaller, and saying "There aren't any filter widgets for this type of field yet. [Learn more]"

**Comment by paulrosenzweig** [comment]
@mazameli I like that solution!

Though, I just messed with it some more and I can't seem to find a use for a field filter without a widget. :/ Even so, I think that's a good solution.

**Comment by paulrosenzweig** [comment]
After the last commit, this is what it looks like when a field has no widgets:
image

**Comment by mazameli** [comment]
> After the last commit, this is what it looks like when a field has no widgets:

Looks good, I'd just move the `There aren't any filter widgets…` text below the `Filter widget type` dropdown.

> Though, I just messed with it some more and I can't seem to find a use for a field filter without a widget.

Simplest case is if you want to be able to connect your SQL question to a dashboard filter, but you don't want/need a filter widget on the SQL question itself. I think there's also the embedding case, where the parameter is strictly for passing in e.g. a user ID to the query, but isn't meant to be user-facing via a widget.

**Review by mazameli — APPROVED** [review]

**Pull request merged**